### PR TITLE
relayer: add unit tests

### LINKED ISSUE #1296: Relayer: add unit tests
We merged the new relayer without having unit tests. We decided to use jest as a test runner. We should achieve basic coverage with the unit tests. We decided to not unit test the inner relaying mechanisms (subclasses of `BaseRelayerService`) because they heavily rely on libraries which are difficult to mock.

### PR DESCRIPTION
**Pull request by manuelwedler**
Closes #1296 

This adds basic unit test coverage for the relayer. As discussed with @GabrielBuragev the actual relaying services are not tested as they rely on libraries and would involve heavy mocking. For the same reason there are no tests for the `service.ts` file.

### CODE REVIEW

**Review by GabrielBuragev — APPROVED** [review]
Thanks for covering this module with unit tests! I like how you cleaned up the messy `BaseRelayerServiceConstructor` type.

I added my changes on top of your work so we can get this merged until you are back from vacation.

Please check my question regarding the `Promise.resolve()` i would really like to know the reason behind it 😄

**Comment by GabrielBuragev** [comment]
@istankovic will include this test setup as a part of the CI here.

File: relayer/tests/unit/common/process.spec.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    Object.defineProperty(process, "ppid", { value: ppid });
+  });
+
+  it("calls exit when pid changes", async () => {
+    const exitSpy = jest.spyOn(process, "exit").mockImplementation();
+
+    killOnParentProcessChange(startPpid);
+    Object.defineProperty(process, "ppid", { value: startPpid + 1 });
+    jest.runOnlyPendingTimers();
+    await Promise.resolve();
```
[the remark this answers is not in the document]
Comment (manuelwedler) [reply relayer/tests/unit/common/process.spec.ts:23]: I think it was to make sure that the promise in the `killOnParentProcessChange` function resolves before continuing here.

[the remark this answers is not in the document]
Comment (GabrielBuragev) [reply relayer/tests/unit/common/process.spec.ts:23]: Ah okay, thanks for explaining. An alternative to that is using `await flushPromises()`.
I think that is more clear and understandable. But this is also fine.

**Pull request merged**